comp/core/secrets/component.go

```diff
@@ -0,0 +1,25 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package secrets decodes secret values by invoking the configured executable command
+package secrets
+
+import (
+	"io"
+)
+
+// team: agent-shared-components
+
+// Component is the component type.
+type Component interface {
+	// Configure the executable command that is used for decoding secrets
+	Configure(command string, arguments []string, timeout, maxSize int, groupExecPerm, removeLinebreak bool)
```

Comment:
Is this function call once? If yes can the content of the method be moved to the constructor of the component?

Reply:
That's the point we discuss last time about `secret` needing the configuration and `config` needing `secret`.

Reply:
Right, this is called by `ResolveSecrets` in pkg/config/config.go. The time when the parameters are available to get passed to this `Configure` method is much later than when the secrets component is constructed.

Reply:
I may be too complicated to do but ideally a component shouldn't have this kind of late initialisation.